### ISSUE TITLE
Poetry run with local files breaks after 0.12.8

### ISSUE THREAD

**Issue by jaysonsantos** [sdispater/poetry#624]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS**:  Any 
- **poetry version**: 0.12.8 

## Issue
After [this commit](https://github.com/sdispater/poetry/commit/48e21915306dd6a8d69b9e94d5609d4d3a8845b2#diff-243e95fabbdb8decc022350c12f83633L304) which got released by 0.12.8, poetry run stopped working with local files.
Before I could do this
```
poetry run ./manage.py 
....
```
and it would work. After a pip install -U poetry==0.12.8 this started happening
```
poetry run ./manage.py 
                                                                       
[FileNotFoundError]                                                   
[Errno 2] No such file or directory: '/Users/jayson/Library/Caches/pypoetry/virtualenvs/cave-py3.6/bin/manage': '/Users/jayson/Library/Caches/pypoetry/virtualenvs/cave-py3.6/bin/manage'  
                                                                                     
run <args> (<args>)...
```
And this will happen both on macos and linux.
[This](https://github.com/sdispater/poetry/commit/48e21915306dd6a8d69b9e94d5609d4d3a8845b2#diff-243e95fabbdb8decc022350c12f83633L304) is the exact lines that changed that made it stop working.
To test it I applied this patch on master
```diff
diff --git a/poetry/utils/env.py b/poetry/utils/env.py
index 2170b04..7175ca6 100644
--- a/poetry/utils/env.py
+++ b/poetry/utils/env.py
@@ -302,6 +302,8 @@ class Env(object):
         Return path to the given executable.
         """
         bin_path = (self._bin_dir / bin).with_suffix(".exe" if self._is_windows else "")
+        if not bin_path.exists():
+            return bin
 
         return str(bin_path)
 
```
And I ran `poetry publish --build -n` and installed the wheel with pip install to test if it worked and it seems that everything is good.

**Comment by festinuz** [comment]
I probably got the same bug

On `poetry:0.12.8`:
```
poetry run atom ./
                                                                                    
[FileNotFoundError]                                                  
[Errno 2] No such file or directory: '/home/topor/.cache/pypoetry/virtualenvs/spider-service-py3.6/bi                       
n/atom': '/home/topor/.cache/pypoetry/virtualenvs/spider-service-py3.6/bin/atom'  
                                                                                    
run <args> (<args>)...
```

It works ok when using `poetry:0.12.7`, so i downgraded for a moment

**Comment by sdispater** [comment]
It should be fixed in the latest `0.12.9` release.
